internal/client/client.go

```diff
@@ -490,8 +502,28 @@ func (c *client) UpdateSharedSource(source *SharedSource) (*SharedSource, error)
 }
 
 // DELETE shared source
-func (c *client) DeleteSharedSource(source *SharedSource) error {
+func (c *client) DeleteSharedSource(source *SharedSource, ctx context.Context) error {
 	url := fmt.Sprintf("%s/v3/pipeline/gateway-route/%s", c.endpoint, source.Id)
 	req := c.newRequest(http.MethodDelete, url, nil)
 	return readBody(c.httpClient.Do(req))
 }
+
+// POST publish pipeline
+func (c *client) PublishPipeline(pipelineId string, ctx context.Context) (*PublishPipeline, error) {
+	url := fmt.Sprintf("%s/v3/pipeline/%s/publish?allow_unconnected_edges=true", c.endpoint, pipelineId)
+	// Because it's a POST, an empty body is required
+	reqBody, err := json.Marshal(struct{}{})
+	if err != nil {
+		return nil, err
+	}
+	req := c.newRequest(http.MethodPost, url, bytes.NewReader(reqBody))
+	resp, err := c.httpClient.Do(req)
```

Comment:
it's just a bit wonky.  The `err` is passed to `readJson` which immediately returns it if it exists.  It's always been this way, I guess it's to save all the callers from doing the same check since most are calling `readJson` anyway.

I see what you mean, though, and the way it works now is confusing.  It's worth an update to fix them all.

Reply:
Done.